### PR TITLE
Added stm32 nucleo_f767zi board support

### PR DESCRIPTION
**Pull request by rolandma**
Tested with Blinky

### CODE REVIEW

File: boards/arm/nucleo_f767zi/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,226 @@
+.. _nucleo_f767zi_board:
+
+ST Nucleo F767ZI
+################
+
+Overview
+********
+
+The STM32 Nucleo-144 boards offer combinations of performance and power that
```
Comment (dbkinder) [inline boards/arm/nucleo_f767zi/doc/index.rst]: Hmmm... the title is for this doc is``ST Nucleo F767ZI`` board, but you're documenting a different board here: ``STM32 Nucleo-144``.  Something's amiss.

File: boards/arm/nucleo_f767zi/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  STM32Cube MCU package.
+- Arm* Mbed Enabled* compliant (only for some Nucleo part numbers)
+
+.. image:: img/nucleo_f767zi.jpg
+   :width: 720px
+   :align: center
+   :height: 720px
+   :alt: Nucleo F767ZI
+
+More information about the board can be found at the `Nucleo F746ZG website`_.
```
Comment (dbkinder) [inline boards/arm/nucleo_f767zi/doc/index.rst]: I suspect this should be `` at the `Nucleo f767zi website`_.``

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: dts/arm/st/f7/stm32f767.dtsi [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2018 Yurii Hamann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/f7/stm32f7.dtsi>
+
+/ {
+	/* 64KB DTCM @ 20000000, 240KB SRAM1 @ 20010000, 16KB SRAM2 @ 2004C000 */
```
Comment (kaidoho) [inline dts/arm/st/f7/stm32f767.dtsi]: This is a copy of the **F746**, but I guess it should be a **F767** 

`/* 128KB DTCM @ 20000000, 368KB SRAM1 @ 20020000, 16KB SRAM2 @ 2007C000 */`

	sram0: memory@20020000 {
		compatible = "mmio-sram";
		reg = <0x20020000 DT_SIZE_K(384)>;
	};

	dtcm: memory@20000000 {
		compatible = "arm,dtcm";
		reg = <0x20000000 DT_SIZE_K(128)>;
	};

File: soc/arm/st_stm32/stm32f7/Kconfig.soc [hunk 4]
```diff
@@ -13,6 +13,9 @@ config SOC_STM32F723XX
 config SOC_STM32F746XX
 	bool "STM32F746XX"
 
+config SOC_STM32F767XX
```
Comment (kaidoho) [inline soc/arm/st_stm32/stm32f7/Kconfig.soc]: For **F767**, please enable the double precision as it is done for **F769**. Could you keep the way the devices are ordered within this file? F756, F767, F769 etc. 

`select CPU_HAS_FPU_DOUBLE_PRECISION`

File: soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+config GPIO_STM32_PORTJ
+	default y
+
+config GPIO_STM32_PORTK
+	default y
+
+endif # GPIO_STM32
+
+config NUM_IRQS
+	default 98
```
Comment (kaidoho) [inline soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx]: `default 110` for **F767**

File: boards/arm/nucleo_f767zi/doc/index.rst (hunk 1 above)
Comment (rolandma) [reply boards/arm/nucleo_f767zi/doc/index.rst]: Following several other Nucleo document format for this Nucleo-144 series board, such as nucleo_f746zg and nucleo_f756zg. I personally have no preference.

File: boards/arm/nucleo_f767zi/doc/index.rst (hunk 2 above)
Comment (rolandma) [reply boards/arm/nucleo_f767zi/doc/index.rst]: Updated.

**Review by rolandma — COMMENTED** [review]
Commit de9821673b:
1: UC3 Title does not follow [subsystem]: [subject] (and should not start with literal subsys:): "Updated Kconfig.defconfig.stm32f767xx and Kconfig.soc for kaidoho's comments."
1: UC5 Title exceeds max length (77>75): "Updated Kconfig.defconfig.stm32f767xx and Kconfig.soc for kaidoho's comments."
1: UC6 Body has no content, should at least have 1 line.

What are the UCx? How to fix them?

**Comment by erwango** [comment]
> What are the UCx? How to fix them?

These are errors reported on the commit messages, if the errors are not enough explicit, you can check https://docs.zephyrproject.org/latest/contribute/#commit-guidelines. Also, you can check the style of other commit messages in zephyr repo.

**Review by erwango — CHANGES_REQUESTED** [review]
Some changes. 

One point about the commits:
When the PR is merged, the commits of the PR are put as is on top of zephyr master branch.
So, the PRs should be clean of any errors, we will not merge commits with fixes or fixes of commits. Fixes done during the review should be integrated directly in the commits.
For this change you're supposed to have only 2 commits, in this order:
- One commit for the dts addition
- One commit for the board addition

So, what I advise you to:
- squash all current commits
- split in two commits (soc and boards)
- for the review, use git fixup and autosquash rebase to integrate the review comments.

To use a fixup commit, put the following in the commit message and then commit:
`fixup! SHA1 of the commit to fix` or `fixup! Title of the commit to fix`
You can add as many fixup commits as you want.
Once you're done,  rebase using --autosquash so git integrate the fixups:
`git rebase -i HEAD~X --autosquash`
X should be higher than then number of commits on your branch

File: dts/arm/st/f7/stm32f767.dtsi [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2018 Yurii Hamann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/f7/stm32f7.dtsi>
```
[the remark this answers is not in the document]
Comment (rolandma) [reply dts/arm/st/f7/stm32f767.dtsi]: Similar as STM32F769.dtsi, the RAM structure is a little different from STM32F746, so I got RAM0 redefined error if I include the stm32f746.dtsi

File: boards/arm/nucleo_f767zi/doc/index.rst (hunk 1 above)
Comment (dbkinder) [reply boards/arm/nucleo_f767zi/doc/index.rst]: Well, that's a bit confusing but I get it now.  The STM32 Nucleo-144 boards are a family while NUCLEO-L496ZG and NUCLEO-L496ZG-P, for example, are specific members of that family of boards.

**Review by dbkinder — APPROVED** [review]
Thanks for explaining about STM32 Nucleo-144 boards vs. specific instances of a board in this family.

File: dts/arm/st/f7/stm32f767.dtsi (hunk 6 above)
[the remark this answers is not in the document]
Comment (erwango) [reply dts/arm/st/f7/stm32f767.dtsi]: Ok, indeed.
We'd need to review the whole stm32f7 series definitions to solve this. Let's keep that aside for now.

**Comment by rolandma** [comment]
Closing this PR because the changes has been re-applied into another PR Add board definition for stm32 nucleo_f767zi #21656 for clean up. All the changes remain the same.
Thanks.

**Pull request closed without merging**